### PR TITLE
Ship a gitignore in Umbraco V9 dotnet new template

### PR DESCRIPTION
**Pull request by warrenbuckley**
This ships the same GitIgnore that Umbraco Cloud will.
Currently issues with the dotnet new template copying this file, investigating...

## Resolution
The problem is with dotnet pack and not dotnet templating engine, see this note on my research
https://github.com/dotnet/templating/issues/823#issuecomment-914549188

### CODE REVIEW

File: build/templates/UmbracoProject/.gitignore [hunk 1]
```diff
@@ -0,0 +1,490 @@
...
+**/umbraco/config/appsettings-schema.json
+
+# This is the no-nodes, installer & upgrader pages from Umbraco
+# Nuget package Umbraco.Cms.StaticAssets will copy them in during dotnet build
+**/umbraco/UmbracoWebsite/
+**/umbraco/UmbracoInstall/
+**/umbraco/UmbracoBackOffice/
+
+# Comment out the line below if you wish to change or add any new templates to PartialView Macros
+**/umbraco/PartialViewMacros/Templates
```
Comment (bergmania) [inline build/templates/UmbracoProject/.gitignore:483]:
```suggestion
**/umbraco/PartialViewMacros/
```

File: build/templates/UmbracoProject/.gitignore [hunk 2]
```diff
@@ -0,0 +1,490 @@
...
+
+# Umbraco Static Assets of Backoffice
+# Nuget package Umbraco.Cms.StaticAssets will copy them in during dotnet build
+**/wwwroot/umbraco/assets/
+**/wwwroot/umbraco/js/
+**/wwwroot/umbraco/lib/
+**/wwwroot/umbraco/views/
```
Comment (bergmania) [inline build/templates/UmbracoProject/.gitignore:490]:
```suggestion
**/wwwroot/umbraco/
```

**Review by bergmania — CHANGES_REQUESTED** [review]
I guess the `/umbraco/Logs/` should also be ignored? And then a few suggestions

**Comment by bergmania** [comment]
Not sure about `/App_Plugins`.. All packages will be added automatically, but if you add some stuff yourself directly, then it will not be source controlled

File: build/templates/UmbracoProject/.gitignore (hunk 1 above)
Comment (warrenbuckley) [reply build/templates/UmbracoProject/.gitignore:483]: But surely I want my own PartialViewMacros to be comitted?

File: build/templates/UmbracoProject/.gitignore (hunk 2 above)
Comment (warrenbuckley) [reply build/templates/UmbracoProject/.gitignore:490]: Is there any other folders in `wwwroot/umbraco/` that I may want to commit?

**Comment by warrenbuckley** [comment]
I think re App_Plugins I would want all of that stuff checked into source control if I install a package

**Comment by warrenbuckley** [comment]
Replied @bergmania let me know your thoughts to my responses and then lets get this wrapped up so Cloud Team has the same gitignore file to work with

File: build/templates/UmbracoProject/.gitignore (hunk 1 above)
Comment (bergmania) [reply build/templates/UmbracoProject/.gitignore:483]: */umbraco/PartialViewMacros/Templates is the one used for macro snippets, not the actual views..

File: build/templates/UmbracoProject/.gitignore (hunk 2 above)
Comment (bergmania) [reply build/templates/UmbracoProject/.gitignore:490]: There should not be that..

Comment (warrenbuckley) [reply build/templates/UmbracoProject/.gitignore:490]: Ok will change this part then

File: build/templates/UmbracoProject/.gitignore (hunk 1 above)
Comment (warrenbuckley) [reply build/templates/UmbracoProject/.gitignore:483]: Yes but you removed the templates part in your suggestion. As I would want to ignore the templates but not my own views that get put in that folder

Comment (bergmania) [reply build/templates/UmbracoProject/.gitignore:483]: your own views would be in `Views/MacroPartials` not here?

**Pull request merged**